lib/auth.jsx

```diff
@@ -159,6 +178,118 @@ export class Auth extends Component {
     this.props.onAuthenticate(username, password);
   };
 
+  onWPLogin = () => {
+    const config = getConfig();
+    this.setupAuthWindow();
+
+    const redirectUrl = encodeURIComponent(config.wpcc_redirect_url);
+    this.authState = `app-${cryptoRandomString(20)}`;
+    const authUrl = `https://public-api.wordpress.com/oauth2/authorize?client_id=${
+      config.wpcc_client_id
+    }&redirect_uri=${redirectUrl}&response_type=code&scope=global&state=${
+      this.authState
+    }`;
+
+    this.authWindow.loadURL(authUrl);
+    this.authWindow.show();
+  };
+
+  setupAuthWindow = () => {
+    const remote = __non_webpack_require__('electron').remote; // eslint-disable-line no-undef
+    const BrowserWindow = remote.BrowserWindow;
+    const protocol = remote.protocol;
+    this.authWindow = new BrowserWindow({
+      width: 640,
+      height: 640,
+      show: false,
+      'node-integration': false,
+    });
+
+    // Register simplenote:// protocol
+    protocol.registerHttpProtocol('simplenote', req => {
+      this.authWindow.loadURL(req.url);
+    });
+
+    this.authWindow.webContents.on('will-navigate', (event, url) =>
+      this.onBrowserNavigate(url)
+    );
+
+    this.authWindow.webContents.on(
+      'did-get-redirect-request',
+      (event, oldUrl, newUrl) => this.onBrowserNavigate(newUrl)
+    );
+  };
+
+  onBrowserNavigate = url => {
+    try {
+      const parsedUrl = new URL(url);
+      // Continue on if the url is not the simplenote protocol
+      if (parsedUrl.protocol !== 'simplenote:') {
+        return;
+      }
+
+      this.processUrl(parsedUrl);
+    } catch (error) {
+      return;
+    }
+  };
+
+  processUrl = url => {
+    const { authorizeUserWithToken, saveWPToken } = this.props;
+    const params = url.searchParams;
+
+    // Display an error message if authorization failed.
+    if (params.get('error')) {
+      const errorCode = params.get('code');
+      if (errorCode === '1') {
+        this.authError(
+          'Please activate your WordPress.com account via email and try again.'
+        );
+      } else {
+        this.authError('An error was encountered while signing in.');
+      }
+
+      return;
+    }
```

Comment:
this is more or less about style and so not a blocker but I think in cases where we are checking a list of possible values it can be helpful to use a `switch` consistently to follow the logic…

```js
switch ( params.get( 'code' ) ) {
  case '1':
    return this.authError( 'Please activate…' );

  default:
    return this.authError( 'An error was encountered…' );
}
```

anyway I don't feel too strongly about this but I see the pattern more often in several different ways and the switch I think is a nice way to keep the intention clear while allowing for more linear expansion (such as when we recognize a second or third distinct error code).

Reply:
> this is more or less about style and so not a blocker but I think in cases where we are checking a list of possible values it can be helpful to use a switch consistently to follow the logic…

I'm cool with that, and would look nicer if we want to check additional error codes in the future.